doc/source/docs/user_guide/io.rst

```diff
@@ -76,6 +81,15 @@ You can also read path objects::
     path_object = pathlib.Path(filename)
     df = geopandas.read_file(path_object)
 
+Using Arrow for faster reading
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For faster data reading, pass ``use_arrow=True`` when using the default pyogrio engine. This can be 2-4 times faster than the default reading behavior and works with all drivers. See `pyogrio.read_dataframe <https://pyogrio.readthedocs.io/en/latest/api.html#pyogrio.read_dataframe>`_ for full details.
+
+It is also possible to enable this by default by setting the environment variable ``PYOGRIO_USE_ARROW=1``.
```

Comment:
Note that the environment variable will also enable writing with arrow, not only reading.

Reference: https://github.com/geopandas/pyogrio/issues/405